src/auth/cognito/awsCognitoAuthSession.ts

```diff
@@ -0,0 +1,31 @@
+import { AuthSession } from "../types/authSession";
+import { Result } from "../types/models/result";
+import { AWSCredentials } from "./types/credentials";
+import { CognitoUserPoolTokens } from "./types/models";
+
+export type CognitoAuthSessionConfig = {
+    isSignedIn: boolean;
+    credentials: Result<AWSCredentials>;
+    userPoolTokens: Result<CognitoUserPoolTokens>;
+    userSub: Result<string>;
+    identityId: Result<string>;
+  };
+  
+  
+  
+  
+  export class AWSCognitoAuthSession implements AuthSession {
```

Comment:
Nit: We should clean up spacing & tabbing before putting this in front of customers. Also had comment on another file about omitting any implementations from the RFC and just focusing on types.